api/handlers/vouchers.go

```diff
@@ -110,6 +110,20 @@ func InsertVoucherHandler(ownerPKeys []crypto.PublicKey) http.HandlerFunc {
 				return
 			}
 
+			// Verify voucher signature chain (manufacturer -> owner transfers)
+			if err := ov.VerifyEntries(); err != nil {
+				slog.Debug("Voucher signature chain verification failed", "guid", ov.Header.Val.GUID[:], "err", err)
+				http.Error(w, "Invalid voucher signature chain", http.StatusBadRequest)
+				return
+			}
+
+			// Verify device certificate chain hash matches header
+			if err := ov.VerifyCertChainHash(); err != nil {
+				slog.Debug("Device cert chain hash verification failed", "guid", ov.Header.Val.GUID[:], "err", err)
+				http.Error(w, "Invalid device certificate chain hash", http.StatusBadRequest)
+				return
+			}
+
```

Comment:
uhm, I think while this is useful to have (?), what we need to have is a way to tell, on insertion, that this OV belongs to us as owner of the OV itself. This doesn't really check that does it?